docs/doxygen/porting.dox

```diff
@@ -0,0 +1,103 @@
+    // Fields necessary for the transport implementations, e.g. a UDP socket descriptor.
+};
+@endcode
+
+@section sntp_porting_gettime Get Time Function
+@brief The coreSNTP library relies on this function to obtain time from system for the operation of tracking timeout
+durations as well as generating time request packet required by the SNTP protocol.
+
+@see @ref SntpGetTime_t
+
+In the scenario that device platform does not have real-world time information (due to device boot-up for example), it is
+acceptable for this function to provide the system-time information that does not match real-world time, because once a
+time information is received in SNTP communication with a time server, the system time can be corrected to match real-time.
+Refer to the next section on how the interface exposed by coreSNTP library to correct system time.
+
+@section sntp_porting_settime Set Time Function
+@brief The coreSNTP library exposes this interface function to notify the device platform about the latest time received
+from a time server as well as the clock drift that the library has calculated of the system time from the server time.
+
+@see @ref SntpSetTime_t
+
+Platforms should implement this function to perform clock disciple operation on the system clock, that is appropriate for
+the clock accuracy needs of the application.
+
+@section sntp_porting_time Authentication Interface
+@brief The coreSNTP library exposes an authentication interface to allow customer-chosen authentication mechanism to be used
+in SNTP communication with time server(s) for security.
+
+@note It is RECOMMENDED to enable authentication in communication with your time server(s) of choice to protect against attacks
+that modify or spoof server responses. The SNTPv4 protocol is flexible to be used with any symmetric-key or asymmetric key
+cryptographic algorithm depending on the support provided by time servers of your choice. For an example of using AES-128-CMAC
+as the authentication algorithm, please refer to [coreSNTP demo in FreeRTOS/FreeRTOS repository](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator).
+
+@see @ref SntpAuthenticationInterface_t
+A port that uses authentication to communicate with time server must implement the following function pointers exposed by the authentication interface:
+ - [Add client authentication code](@ref SntpGenerateAuthCode_t): A function to generate and append authentication data for client to be validated
+ by the time server. The authentication data SHOULD be generated with the @ref SNTP_PACKET_BASE_SIZE on the SNTP request packet bytes serialized by the
+ coreSNTP library and appended to the same buffer. This function should also indicate the number of authentication bytes appended to the library
+ through an output parameter, so that the library knows about the total size of the SNTP packets that will be communicated with the time server
+ that supports the chosen authentication.
```

Comment:
```suggestion
 coreSNTP library and appended to the same buffer. This function should also return the number of authentication bytes appended to the library
 through an output parameter, so that the library knows about the total size of the SNTP packet.
```